node/src/types/appendable_block.rs

```diff
@@ -61,49 +61,65 @@ impl AppendableBlock {
         self.total_size
     }
 
+    /// Attempts to add a transfer to the block; returns an error if that would violate a validity
+    /// condition.
+    pub(crate) fn add_transfer(
+        &mut self,
+        hash: DeployHash,
+        deploy_info: &DeployInfo,
+    ) -> Result<(), AddError> {
+        if self.deploy_and_transfer_set.contains(&hash) {
+            return Err(AddError::Duplicate);
+        }
+        if !deploy_info
+            .header
+            .is_valid(&self.deploy_config, self.timestamp)
+        {
+            return Err(AddError::InvalidDeploy);
+        }
+        if self.has_max_transfer_count() {
+            return Err(AddError::TransferCount);
+        }
+        self.transfer_hashes.push(hash);
+        self.deploy_and_transfer_set.insert(hash);
+        Ok(())
+    }
+
     /// Attempts to add a deploy to the block; returns an error if that would violate a validity
     /// condition.
```

Comment:
Maybe add to the comment that this must not be called with a transfer, and it cannot check whether it's a transfer.

This actually makes me wonder whether we should have two wrapper types of `DeployHash` to make the type system prevent us from mixing up transfers and other deploys?

Reply:
I was thinking of it, actually, only I'm afraid of how far reaching this change would have to be. But maybe I should just bite the bullet and do it?

Reply:
Although... Wouldn't that affect externally facing APIs, too?

Reply:
The wrapper type could be used internally only; but I agree, it would be a big change.